File: backend/app/app/models/gift.py
from sqlalchemy import Column, Float, Integer, String

from .session import Base


class Gift(Base):
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    url = Column(String, nullable=True)
    price = Column(Float, nullable=True)
    buy_price = Column(Float, nullable=True)
    recipient = Column(String, index=True)
    giver = Column(String, index=True, nullable=True)

File: frontend/src/app/gift.ts
export interface Gift {
    id: number;
    name: string;
    url?: string;
    price?: number;
    buy_price?: number;
    recipient: string;
    giver?: string;
  }

export class Page<ClassType> {
    items: ClassType[] = [];
    total: number = 0;
    page: number = 0;
    size: number = 0;
  }